File: CS/TreeListHorizontalScrolling/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;

namespace TreeListHorizontalScrolling {
    public partial class Form1 : Form {


        List<Customer> list = new List<Customer>();
        TreeListExtensionScrolling Extension;
        
        public Form1() {
            InitializeComponent();

            textEdit1.KeyDown += textEdit1_KeyDown;
            treeList1.LeftCoordChanged += treeList1_LeftCoordChanged;

            for (int i = 0; i < 10; i++)
                list.Add(new Customer() { Zero = "1", One = "1", Two = "1", Three = "1", Four = "1", Five = "1", Six = "1" });
            treeList1.DataSource = list;

            Extension = new TreeListExtensionScrolling(treeList1);

            treeList1.Columns[0].Fixed = DevExpress.XtraTreeList.Columns.FixedStyle.Left;
            treeList1.Columns[3].Fixed = DevExpress.XtraTreeList.Columns.FixedStyle.Left;
        }

        void treeList1_LeftCoordChanged(object sender, EventArgs e) {
            textEdit1.Text = Extension.UpdateIndexInEditor().ToString();
        }

        void textEdit1_KeyDown(object sender, KeyEventArgs e) {
            if (e.KeyCode == Keys.Enter)
                if (textEdit1.Text != string.Empty)
                    Extension.ScrollTo(Convert.ToInt32(textEdit1.Text));
        }

        private void Form1_Load(object sender, EventArgs e) {
            Extension.UpdateFixedColumnsCount();
            textEdit1.Text = Extension.UpdateIndexInEditor().ToString();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            Extension.ScrollBackward();
        }

        private void simpleButton2_Click(object sender, EventArgs e) {
            Extension.ScrollForward();
        }
    }
}

File: VB/TreeListHorizontalScrolling/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms

Namespace TreeListHorizontalScrolling
	Partial Public Class Form1
		Inherits Form


		Private list As New List(Of Customer)()
		Private Extension As TreeListExtensionScrolling

		Public Sub New()
			InitializeComponent()

			AddHandler textEdit1.KeyDown, AddressOf textEdit1_KeyDown
			AddHandler treeList1.LeftCoordChanged, AddressOf treeList1_LeftCoordChanged

			For i As Integer = 0 To 9
				list.Add(New Customer() With {.Zero = "1", .One = "1", .Two = "1", .Three = "1", .Four = "1", .Five = "1", .Six = "1"})
			Next i
			treeList1.DataSource = list

			Extension = New TreeListExtensionScrolling(treeList1)

			treeList1.Columns(0).Fixed = DevExpress.XtraTreeList.Columns.FixedStyle.Left
			treeList1.Columns(3).Fixed = DevExpress.XtraTreeList.Columns.FixedStyle.Left
		End Sub

		Private Sub treeList1_LeftCoordChanged(ByVal sender As Object, ByVal e As EventArgs)
			textEdit1.Text = Extension.UpdateIndexInEditor().ToString()
		End Sub

		Private Sub textEdit1_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs)
			If e.KeyCode = Keys.Enter Then
				If textEdit1.Text <> String.Empty Then
					Extension.ScrollTo(Convert.ToInt32(textEdit1.Text))
				End If
			End If
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Extension.UpdateFixedColumnsCount()
			textEdit1.Text = Extension.UpdateIndexInEditor().ToString()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			Extension.ScrollBackward()
		End Sub

		Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton2.Click
			Extension.ScrollForward()
		End Sub
	End Class
End Namespace